geode-redis/src/main/java/org/apache/geode/redis/internal/executor/pubsub/PublishExecutor.java

```diff
@@ -27,12 +30,34 @@
   @Override
   public RedisResponse executeCommand(Command command,
       ExecutionHandlerContext context) {
-    List<byte[]> args = command.getProcessedCommand();
-
-    long publishCount =
-        context.getPubSub().publish(getDataRegion(context), args.get(1), args.get(2));
-
-    return RedisResponse.integer(publishCount);
+    context.getBackgroundExecutor().submit(new PublishingRunnable(context, command));
+    return null;
   }
 
+  public static class PublishingRunnable implements Runnable {
+
+    private final ExecutionHandlerContext context;
+    private final Command command;
+
+    public PublishingRunnable(ExecutionHandlerContext context, Command command) {
+      this.context = context;
+      this.command = command;
+    }
+
+    @Override
+    public void run() {
+      List<byte[]> args = command.getProcessedCommand();
+      byte[] channelName = args.get(1);
+      byte[] message = args.get(2);
+      try {
+        long publishCount =
+            context.getPubSub()
+                .publish(context.getRegionProvider().getDataRegion(), channelName, message);
+        ByteBuf response = Coder.getIntegerResponse(context.getByteBufAllocator(), publishCount);
```

Comment:
I'd prefer to create a `RedisResponse` so that we keep netty specifics out of the executors.

Reply:
Done. I also found one other place (in AbstractSubscription) that was not using RedisResponse. I change it also to do so and deleted the old writeToChannel(ByteBuf) which was no longer used after these changes.